datastream/dataset.py

```diff
@@ -188,7 +234,11 @@ def combine(datasets):
         )
 
     @staticmethod
-    def zip(datasets):
+    def zip(datasets: List[Dataset]) -> Dataset:
+        '''
+        Zip multiple datasets together so that examples with matching indices
```

Comment:
* What happens when they don't match?
* Is there any requirements on the indices for the input datasets?

Reply:
Like the standard zip it has the length of the shortest. I can add this